### PR TITLE
[TB] Support writing new style scalar

### PR DESCRIPTION
**Pull request by cryptopic**
Summary:
New style vs old style
https://github.com/tensorflow/tensorboard/blob/b306651ab5baa6d362326464a703463b4683b265/tensorboard/data_compat.py#L49-L53

Writing in new style can help avoid the cost of migration
https://github.com/tensorflow/tensorboard/blob/b306651ab5baa6d362326464a703463b4683b265/tensorboard/data_compat.py#L46

----

Test Plan:
buck run caffe2/test:tensorboard

 ---

Differential Revision: D26879076

### CODE REVIEW

File: torch/utils/tensorboard/summary.py [hunk 1]
```diff
@@ -239,15 +239,30 @@ def scalar(name, scalar, collections=None):
       tensor: A real numeric Tensor containing a single value.
       collections: Optional list of graph collections keys. The new summary op is
         added to these collections. Defaults to `[GraphKeys.SUMMARIES]`.
+      new_style: Whether to use new style (tensor field) or old style (simply_value
```
Comment (edward-io) [inline torch/utils/tensorboard/summary.py:242]: nit: simple_value

File: torch/utils/tensorboard/writer.py [hunk 2]
```diff
@@ -321,7 +323,8 @@ def add_scalar(self, tag, scalar_value, global_step=None, walltime=None):
             global_step (int): Global step value to record
             walltime (float): Optional override default walltime (time.time())
               with seconds after epoch of event
-
+            new_style (boolean): Whether to use new style (tensor field) or old
+              style (simply_value field)
```
Comment (edward-io) [inline torch/utils/tensorboard/writer.py:327]: nit: simple_value

**Review by edward-io — APPROVED** [review]
Looks good to me. Have you tested that the new-style can be read and visualized successfully using the OSS TB version? Also, should we let users know what the difference is between the two? Like which one they should pick.

**Comment by cryptopic** [comment]
> Looks good to me. Have you tested that the new-style can be read and visualized successfully using the OSS TB version? 

Good call, I tried loading OSS Tensorboard with new style file, and was able to correctly see results

> Also, should we let users know what the difference is between the two? Like which one they should pick.

I intentionally avoided making suggestions, because the profiling result showing 50% latency win is our specific use case on our internal stack with older version of TB, the win might not be there for OSS users. Although I can add comment mentioning that new_style **could** lead to latency win.

**Pull request closed without merging**